### ISSUE TITLE
Questions about interact.py

### ISSUE THREAD

**Issue by GlitchBox** [huggingface/transfer-learning-conv-ai#124]
Hi, 
Can anyone explain why the code snippet ([interact.py](https://github.com/huggingface/transfer-learning-conv-ai/blob/d4c760731e8f062abd3d61ccdc8fa29ccb5e3485/interact.py#L72)) (underlined in red) was necessary? 

![Screenshot from 2022-09-15 11-58-25](https://user-images.githubusercontent.com/34965899/190329464-a10809fc-bb38-473d-9331-2176970ab88a.png)

As far as I know the logits returned by **OpenAIGPTLMHeadModel** is of the following form : (batch_size, sequence_length, vocabulary size). 

Why was only the last token in the output sequence considered as the predicted next token?
Moreover, why do we have to iteratively generate an output text when the model itself returns a full sequence and not just a single token?
